lib/ii.c

```diff
@@ -11582,13 +11582,17 @@ grn_ii_select_data_find_phrase(grn_ctx *ctx,
   token_info **tip;
   token_info **tis = data->token_infos;
   token_info **tie = tis + data->n_token_infos;
+  token_info *must_last_ti = NULL;
   uint32_t n_tokens_in_same_pos = 0;
   for (tip = tis; ; tip++) {
     if (tip == tie) { tip = tis; }
     data->token_info = *tip;
     if (data->token_info->phrase_id != phrase_id) {
       continue;
```

Comment:
Do we need to reset here to avoid returning wrong phrase's token info?

```suggestion
      must_last_ti = NULL;
      continue;
```

Reply:
I think no.
`must_last_ti` is updated or assigned to `data->token_info->phrase_id` only when `data->token_info->phrase_id == phrase_id`, and `phrase_id` is never updated in this function.
So wrong phrase's token info is never assigned to `data->token_info->phrase_id`.

Also, if resetting `must_last_ti` at this line, this function don't work as expected in the case below:

`phrase_id` specified with argument: 1
`token1`: phrase_id = 1, must_last = true
`token2`: phrase_id = 2
`token3`: phrase_id = 1, must_last = false

|Processing token|`must_last_ti`|
|-|-|
|`token1`|`token1`|
|`token2`|`NULL`|
|`token3`|`NULL`|

When processing `token3`, `must_last_ti` should be `token1`, but `NULL`.

But... we never get this sort order because the tokens of same `phrase_id` are continuous...

Reply:
I see.